filesystem/fs_interface.go

```diff
@@ -0,0 +1,14 @@
+package filesystem
+
+type Client interface {
+	Open() File
```

Comment:
It is probably better to return io.ReadCloser instead of File, so that code is modular.